src/server/routes/carts.js

```diff
@@ -16,6 +21,10 @@ export function createCartsRouter({ cartRepo }) {
     //add bag, so a cartItem, to cart by userid
     router.post("/:userId/bags", async (req, res) => {
         const userId = parseInt(req.params.userId);
+         //verify the user actually exists
+        if (!userRepo.getUserById/userId){
+            return res.status(404).json({ error: `User ${userId} not found!` });
```

Comment:
Typo: there's a slash in `getUserById/userId`.

I'm ok with keeping these userId checks, but personally I think they aren't required, because reference key constraints are already enforced by the database. Also, there are a lot of reference keys to check in the app. It's hard to keep track of them all. You missed some about the estId.

To be consistent, I would let the database handle these references constraints.